core/cat/routes/upload.py

```diff
@@ -115,9 +119,81 @@ async def upload_memory(
     # access cat instance
     ccat = request.app.state.ccat
 
-    raise HTTPException(
-        status_code = 422,
-        detail = {
-            "error": "Unable to reach the link",
-        },
+    # Check file mime type
+    content_type = mimetypes.guess_type(file.filename)[0]
+
+    # check if MIME type of uploaded file is supported
+    if content_type != "application/json":
+        raise HTTPException(
+            status_code=422,
+            detail={
+                "error": f'MIME type {file.content_type} not supported. Admitted types: application/json'}
+        )
+
+    # Create temporary file
+    temp_file = tempfile.NamedTemporaryFile(dir="/tmp/", delete=False)
```

Comment:
I started to see these lines of code in several places, probably could be good start to move in a proper service these action in order to call directly a specific method and unify the logic in a one place.

Reply:
@nicola-corbellini what do you think about this :smiley:  ?

Reply:
Yes, I agree. It may be a good solution.
Yet, if the tmp file is truly needed here is still an open question to me honestly 😅

Reply:
@nicola-corbellini @giovannialbero1992 honestly I don't remember why we save the files on disk, it may be in fact unnecessary

Reply:
Alright @pieroit , I'll remove it then in the next commit. 
Also, as @giovannialbero1992 suggested, I could make a private method in the RabbitHole for mime type checking along with the `upload_memory` method we discussed.

Reply:
Sounds great!